extensions/pkg/controller/infrastructure/reconciler.go

```diff
@@ -190,6 +203,11 @@ func (r *reconciler) restore(ctx context.Context, logger logr.Logger, infrastruc
 		return reconcile.Result{}, err
 	}
 
+	if err := r.validateConfig(ctx, infrastructure); err != nil {
```

Comment:
If the users changes both `spec.seedName` and `spec.provider.infrastructureConfig.networks.cloudNAT.natIPNames` (to something non-existing or not available), then the first operation to actually detect the error will be `restore`. It's true that for most of our extensions, the restore operation for infrastructure just unmarshals a raw state and therefore restore wouldn't fail without the check, but this is a special case because of the way we use terraform. In general, restore might just do a rather normal reconcile (see e.g. how it's implemented for kubevirt), and therefore the config check is relevant for this operation.